Managers/UTMVirtualMachine+Drives.m

```diff
@@ -73,19 +73,18 @@ - (BOOL)ejectDrive:(UTMDrive *)drive force:(BOOL)force error:(NSError * _Nullabl
 }
 
 - (BOOL)changeMediumForDrive:(UTMDrive *)drive url:(NSURL *)url error:(NSError * _Nullable __autoreleasing *)error {
+    [url startAccessingSecurityScopedResource];
     NSData *bookmark = [url bookmarkDataWithOptions:0
                      includingResourceValuesForKeys:nil
                                       relativeToURL:nil
                                               error:error];
+    [url stopAccessingSecurityScopedResource];
     if (!bookmark) {
         if (error) {
             *error = [NSError errorWithDomain:kUTMErrorDomain code:-1 userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"Failed create bookmark.", "UTMVirtualMachine+Drives")}];
         }
         return NO;
     }
-    if (![self checkSandboxAccess:url error:error]) {
-        return NO;
-    }
     [self.viewState setBookmark:bookmark path:url.path forRemovableDrive:drive.name persistent:NO];
```

Comment:
You're temporarily saving the bookmark with `persistent:NO` to update the view state. However it is saved across UTM restarts anyway. Maybe you could rename the flag to make the code more readable?

Reply:
In line 104 you are calling the `setBookmark` with `persistent:YES`, disregarding the `persistent` parameter passed, and above that in line 101 you are passing `persistent` to `securityScoped`. I don't understand what's happening here.